pytest_fauxfactory.py

```diff
@@ -29,6 +32,27 @@ def gen_string(items=None, str_type=None, *args, **kwargs):
         item += 1
 
 
+class GenString:
+    __slots__ = ['name', 'args', 'kwargs']
+
+    def __init__(self, name=None, args=None, kwargs=None):
+        if name is None:
+            name = fauxfactory.gen_choice(STRING_TYPES)
+        if args is None:
+            args = ()
+        if kwargs is None:
+            kwargs = {}
+
+        # todo type and value inspections on name args kwargs
+        self.name = name
+        self.args = args
+        self.kwargs = kwargs
```

Comment:
For this use case, `attrs` PyPI library may be useful.

Reply:
Hi @mfalesni , thanks for looking this is simply a draft for the moment that show the main idea, if it's OK, than we can investigate should we use third party library for this , looked at attrs nice lib to have  :).